www/front_src/src/Resources/Details/tabs/Graph/models.ts

```diff
@@ -4,39 +4,75 @@ import { find, propEq } from 'ramda';
 import { timeFormat, dateFormat } from '@centreon/ui';
 
 import {
-  labelLast24h,
-  labelLast7Days,
+  label24H,
+  label7D,
+  label31D,
+  label31Days,
   labelLast31Days,
+  label7Days,
+  labelLast7Days,
+  label24Hours,
+  labelLast24Hours,
 } from '../../../translatedLabels';
 
 export type TimePeriodId = 'last_24_h' | 'last_7_days' | 'last_31_days';
 
 export interface TimePeriod {
   id: TimePeriodId;
+  compactName: string;
   name: string;
+  largeName: string;
   getStart: () => Date;
   dateTimeFormat: string;
   timelineEventsLimit: number;
 }
 
+export interface CustomTimePeriod {
+  start: Date;
+  end: Date;
+  xAxisTickFormat?: string;
+  timelineLimit?: number;
+}
+
+export interface StoredCustomTimePeriod {
+  start: string;
+  end: string;
+}
+
+export enum CustomTimePeriodProperties {
+  start = 'start',
+  end = 'end',
+}
+
+export interface ChangeCustomTimePeriodProps {
+  date: Date;
+  property: CustomTimePeriodProperties;
```

Comment:
```suggestion
  property: CustomTimePeriodProperty;
```

Use singular for enums, as recommanded here https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-1.1/4x252001(v=vs.71)?redirectedfrom=MSDN